metakb/transform/civic.py

```diff
@@ -0,0 +1,488 @@
+        :return A list of Gene Descriptor
+        """
+        found_match = False
+        for query_str in [f"ncbigene:{gene['entrez_id']}", gene['name']] + gene['aliases']:  # noqa: E501
+            gene_norm_resp = \
+                self.gene_query_handler.search_sources(query_str, incl="hgnc")
+            if gene_norm_resp['source_matches']:
+                if gene_norm_resp['source_matches'][0]['match_type'] != 0:
+                    found_match = True
+                    break
+
+        if found_match:
+            gene_descriptor = [schemas.GeneDescriptor(
+                id=f"civic:gid{gene['id']}",
+                label=gene['name'],
+                description=gene['description'] if gene['description'] else None,  # noqa: E501
+                value=schemas.Gene(gene_id=gene_norm_resp['source_matches'][0]['records'][0].concept_id),  # noqa: E501
+                alternate_labels=gene['aliases']
+            ).dict()]
+        else:
+            gene_descriptor = []
+
+        return gene_descriptor
+
+    def _get_disease_descriptors(self, disease):
+        """Return A list of Disease Descriptors.
+        :param dict disease: A CIViC disease record
+        :return: A list of Disease Descriptors.
+        """
+        if not disease['doid']:
+            logger.warning(f"CIViC {disease['id']} has null DOID.")
+            return []
+
+        doid = f"doid:{disease['doid']}"
+        disease_norm_resp = self.disease_query_handler.search_groups(doid)
+
+        display_name = disease['display_name']
+        if disease_norm_resp['match_type'] == 0:
+            disease_norm_resp = \
+                self.disease_query_handler.search_groups(display_name)
+
+        if disease_norm_resp['match_type'] == 0:
+            logger.warning(f"{doid}: {display_name} not found in Disease "
+                           f"Normalization normalize.")
+            return []
+
+        disease_norm_id = \
+            disease_norm_resp['value_object_descriptor']['value']['disease_id']
+
+        if disease_norm_id.startswith('ncit:'):
+            disease_descriptor = schemas.ValueObjectDescriptor(
+                id=f"civic:did{disease['id']}",
+                type="DiseaseDescriptor",
+                label=display_name,
+                value=schemas.Disease(disease_id=disease_norm_id),
+            ).dict()
+        else:
+            # TODO: Should we accept other disease_ids other than NCIt?
+            logger.warning("Could not NCIt ID using Disease Normalization "
+                           f"for {doid} or {display_name}.")
+            return []
+
+        return [disease_descriptor]
+
+    def _get_therapy_descriptors(self, drugs):
+        """Return a list of Therapy Descriptors.
+        :param dict drugs: Drugs for a given evidence_item
+        :return: A list of Therapy Descriptors
+        """
+        therapies = list()
+        if len(drugs) != 1:
+            return therapies
```

Comment:
is there any way to combine this check with the stuff on line 93-94 or do they cover different angles?